core/src/main/java/org/apache/lucene/queries/NotCacheable.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.lucene.queries;
+
+/**
+ * Interface used to denote that a given instance cannot be cached.
+ */
+public interface NotCacheable {}
```

Comment:
I wonder why we would do that and not turn it around and cache the actual result of our query per segment. I mean it would simplify all the stuff and we had a stateless query. The uid query would just calculate the bitset for one slice in when the scorer is pulled and we would just cache things in the query cache such that the next request can reuse it? this would also prevent us from holding on to the `DocIDSet[]` in the query and we'd not need any cache key? All we need is the slice ID here afaik?